android/canonical/app/src/main/java/com/google/samples/quickstart/canonical/MapsFragment.kt

```diff
@@ -51,27 +70,70 @@ class MapsFragment : Fragment() {
         }
     }
 
+    private fun initPlaces() {
+        context?.let { Places.initialize(it, getString(R.string.google_maps_key)) }
+        placesClient = context?.let { Places.createClient(it) }!!
+    }
+
+    private fun setPlacesSearchBias() {
+        // Search nearby result
+        currentLatLng?.let {
+            Log.i(TAG, "currentLatLng")
+            autocompleteFragment.setLocationBias(
+                RectangularBounds.newInstance(
+                    LatLng(currentLatLng!!.latitude - 1, currentLatLng!!.longitude - 1),
+                    LatLng(currentLatLng!!.latitude + 1, currentLatLng!!.longitude + 1)
+                ))
+        }
+    }
+
     private fun setUpMap() {
         if (checkSelfPermission(context as Activity,
                 Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
             requestPermissions(arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUEST_CODE)
             return
         }
 
+        // Add and adjust the position of MyLocation button.
         map.isMyLocationEnabled = true
+        map.setPadding(0,(getString(R.string.padding_ratio).toFloat() * autocompleteLayout.height).toInt(),0,0)
```

Comment:
This should be a constant value, not a string resource. Consider if this can be a dimension resource https://developer.android.com/guide/topics/resources/more-resources#Dimension. Android has dp dimensions that scale based on screen density.